Swift/BlurHashDecode.swift

```diff
@@ -1,21 +1,63 @@
+#if canImport(SwiftUI)
+import SwiftUI
+
+extension Image {
+
+#if canImport(UIKit)
+    public init?(blurHash: String, size: CGSize, punch: Float = 1) {
+        guard let uiImage = UIImage(blurHash: blurHash, size: size, punch: punch) else { return nil }
+        self.init(uiImage: uiImage)
+    }
+#elseif canImport(AppKit)
+    public init?(blurHash: String, size: CGSize, punch: Float = 1) {
+        guard let nsImage = NSImage(blurHash: blurHash, size: size, punch: punch) else { return nil }
+        self.init(nsImage: nsImage)
+    }
+#endif
+    
+}
+
+#endif
+
+#if canImport(UIKit)
 import UIKit
 
 extension UIImage {
     public convenience init?(blurHash: String, size: CGSize, punch: Float = 1) {
+        
+        guard let cgImage = CGImage.from(blurHash: blurHash, size: size, punch: punch) else { return nil }
+
+        self.init(cgImage: cgImage)
+    }
+}
+#elseif canImport(AppKit)
+import AppKit
+
+extension NSImage {
+    public convenience init?(blurHash: String, size: CGSize, punch: Float = 1) {
+        guard let cgImage = CGImage.from(blurHash: blurHash, size: size, punch: punch) else { return nil }
+
+        self.init(cgImage: cgImage, size: .zero)
+    }
+}
+#endif
+
+extension CGImage {
+    static func from(blurHash: String, size: CGSize, punch: Float = 1) -> CGImage? {
```

Comment:
Could this be `public convenience init?(blurHash: String, size: CGSize, punch: Float = 1)`?

Would be in line with what is provided by the other classes.

Reply:
I also wanted to use that, but apparently for CG and CF types it's not possible, XCode gives the following error and it will refuse to build:
"Convenience initializers are not supported in extensions of CF types"